src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void updateFruitStorage(FruitTransaction fruitTransaction,
+                                   FruitStorageDao fruitStorageDao) {
+        fruitStorageDao.decreaseQuantity(fruitTransaction);
+    }
```

Comment:
According to the checklist, handling Purchase operation should include a check to ensure that the result balance is positive before saving it in the Storage. If the balance is negative, you should throw a RuntimeException. Currently, the `updateFruitStorage` method does not perform this check.